plugin/src/test/java/io/jenkins/plugins/analysis/core/steps/WarningChecksPublisherITest.java

```diff
@@ -0,0 +1,104 @@
+package io.jenkins.plugins.analysis.core.steps;
+
+import org.junit.Test;
+
+import hudson.model.FreeStyleProject;
+import hudson.model.Run;
+
+import io.jenkins.plugins.analysis.core.testutil.IntegrationTestWithJenkinsPerTest;
+import io.jenkins.plugins.checks.api.ChecksAnnotation.ChecksAnnotationBuilder;
+import io.jenkins.plugins.checks.api.ChecksAnnotation.ChecksAnnotationLevel;
+import io.jenkins.plugins.checks.api.ChecksConclusion;
+import io.jenkins.plugins.checks.api.ChecksDetails;
+import io.jenkins.plugins.checks.api.ChecksDetails.ChecksDetailsBuilder;
+import io.jenkins.plugins.checks.api.ChecksOutput;
+import io.jenkins.plugins.checks.api.ChecksOutput.ChecksOutputBuilder;
+import io.jenkins.plugins.checks.api.ChecksStatus;
+
+import static io.jenkins.plugins.analysis.core.assertions.Assertions.*;
+
+public class WarningChecksPublisherITest extends IntegrationTestWithJenkinsPerTest {
+    private static final String OLD_REPORT_FILE = "checkstyle.xml";
+    private static final String NEW_REPORT_FILE = "checkstyle1.xml";
+
+    /**
+     * Verifies that {@link WarningChecksPublisher} constructs the {@link ChecksDetails} correctly
+     * with only new issues.
+     */
+    @Test
+    public void shouldCreateChecksDetailsWithNewIssuesAsAnnotations() {
+        FreeStyleProject project = createFreeStyleProjectWithWorkspaceFiles(OLD_REPORT_FILE);
+        enableCheckStyleWarnings(project);
+
+        // trigger a build using the report file with 4 issues
+        Run<?, ?> run = buildSuccessfully(project);
```

Comment:
If you have 2 builds use a variable `reference` for the first one and `run` or `build` for the second one (to make it consistent to most of my other tests)